cdap-explore/src/main/java/co/cask/cdap/explore/service/ExploreServiceUtils.java

```diff
@@ -496,4 +513,20 @@ private static File updateHiveConfFile(File confFile, File tempDir) {
     }
     return newHiveConfFile;
   }
+
+  public static boolean isSparkAvailable() {
+    try {
+      // SparkUtils.locateSparkAssemblyJar() throws IllegalStateException if it is not able to locate spark jar
+      return System.getenv(Constants.SPARK_HOME) != null && SparkUtils.locateSparkAssemblyJar() != null;
```

Comment:
`locateSparkAssemblyJar` never returns null. Besides why need to check for `SPARK_HOME` existence? I think this method only get called on the master box, right? The `locateSparkAssemblyJar` already has the logic to check SPARK_HOME.